### ISSUE TITLE
A loss function for calculating categorical cross-entropy from the softmax probabilities

### ISSUE THREAD

**Issue by EmielBoss** [tensorflow/tensorflow#29081]
**System information**
- TensorFlow version (you are using): 1.12
- Are you willing to contribute it (Yes/No): No

**Describe the feature and the current behavior/state.**
I wish for a loss function that calculates categorical cross-entropy based on softmax predictions. As far as I have seen, there are only function that work with the logits. Why is there no way to use the softmax scores?

**Will this change the current api? How?**
Yes, a function will be added.

**Who will benefit with this feature?**
People who wish to conceptually separate the model output and loss calculation. Also, in this way I do not have to change the model in order to use it for prediction.

**Comment by lufol** [comment]
Hi, did you already look at `tf.keras.losses.CategoricalCrossentropy`? It seems to work with both `from_logits = True/False`.
As last layer of your network you still want to use softmax right?

**Comment by EmielBoss** [comment]
> Hi, did you already look at `tf.keras.losses.CategoricalCrossentropy`? It seems to work with both `from_logits = True/False`.
> As last layer of your network you still want to use softmax right?

Completely missed that one, thanks! And yes, I would like to be able to complete my model with a softmax layer.
